api/config.go

```diff
@@ -60,14 +61,15 @@ var (
 	GlobalTLSCertsManager *xcerts.Manager
 )
 
-// getK8sSAToken assumes the plugin is running inside a k8s pod and extract the current service account from the
-// /var/run/secrets/kubernetes.io/serviceaccount/token file
-func getK8sSAToken() string {
-	dat, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/token")
-	if err != nil {
+// getK8sSAToken assumes the plugin is running inside a k8s pod and gets the token directly from IdP as id_token
+// if id_token is valid token for k8s, then user will have access as described in k8s documentation:
+// https://kubernetes.io/docs/reference/access-authn-authz/authentication/#openid-connect-tokens
+func getK8sSAToken(token interface{}) string {
```

Comment:
You may also want to pass in `*xoauth2.Token` and check for the `id_token` in this method. That prevents using `interface{}` in the prototypes. If you want to keep it like this, then consider using `any` instead of `interface{}`.